### PR TITLE
MAV-439: Write a CQRS reactor shell

### PR DESCRIPTION
**Pull request by Faro808**
Tests rely on PostgreSQL.

### CODE REVIEW

File: cqrs-postgres/src/reactor.rs [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+        query: &str,
+        since: Since,
+        params: &[Box<dyn ToSql>],
+    ) -> Result<Vec<RawEvent>, postgres::Error> {
+        let last_sequence = match since {
+            Since::BeginningOfStream => 0,
+            Since::Event(x) => x.get(),
+        } as i64;
+
+        let trans =
```
Comment (ajgajg1134) [inline cqrs-postgres/src/reactor.rs]: Is there a specific reason to have a transaction here as opposed to just making the query by itself (Which if it's a single query is treated as a transaction I believe?)

File: cqrs-postgres/src/reactor.rs [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+                     FROM events \
+                     WHERE event_id > $1 AND (FALSE",
+                );
+
+                let mut param_count = 1;
+
+                for predicate in aggregate_predicates {
+                    match &predicate.event_types {
+                        EventTypesPredicate::SpecificEventTypes(event_types) => {
+                            write!(
```
Comment (ajgajg1134) [inline cqrs-postgres/src/reactor.rs]: What macro is this?

File: cqrs-postgres/src/reactor.rs [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+            run: AtomicBool::new(true),
+        }
+    }
+
+    pub fn stop_reaction(&self) {
+        self.run.store(false, Ordering::Relaxed);
+    }
+
+    pub fn start_reaction<R: Reaction>(&self, _reaction: R) {
+        let conn = self.pool.get().unwrap();
```
[the remark this answers is not in the document]
Comment (Faro808) [reply cqrs-postgres/src/reactor.rs]: Neither unwrap nor expect should be in production code, as they both panic. Unless we just want to rely on the k8s restart. I'd prefer to add proper error handling, using `?` and `Result`. Perhaps I wasn't ready for a PR, yet.

File: cqrs-postgres/src/reactor.rs (hunk 2 above)
Comment (Faro808) [reply cqrs-postgres/src/reactor.rs]: "Write formatted data into a buffer."

File: cqrs-postgres/src/reactor.rs (hunk 1 above)
Comment (Faro808) [reply cqrs-postgres/src/reactor.rs]: IIUMC ("If I Understood Marcus, Correctly"): When we create the `Transaction` with `read_only(true))?;`, PostgreSQL can optimize the transactions better.

Comment (neoeinstein) [reply cqrs-postgres/src/reactor.rs]: This is going to go away for some of these, as the single query can be the complete transaction instead.

**Pull request closed without merging**